# Define the goal for this command
uint32  commandType
uint32  actionOnDone
float64 numParam1
float64 numParam2
string  strParam1 
string  comment


# Command Types

uint32  FF_CMD_CLEAR_COMMANDS=1       # Clear all commands (good idea on start of activities)
uint32  FF_CMD_WAIT_IN_SECONDS=2      # stop execution of commands for this delay in seconds
uint32  FF_CMD_CLEAR_IN_PROGRESS=9    # Clear queue special internal command

uint32  FF_CMD_FOLLOW_FIDUCIAL=101    # Approach a fiducial up to a preset distance
uint32  FF_CMD_STOP_MOVEMENT=102
uint32  FF_CMD_DRIVE_FORWARD=103      # Drive forward for the specified time at the current drive_rate
uint32  FF_CMD_DRIVE_REVERSE=104      # Drive reverse for the specified time at the current drive_rate
uint32  FF_CMD_ROTATE_LEFT=105        # Rotate left   for the specified time at the current rotate_rate
uint32  FF_CMD_ROTATE_RIGHT=106       # Rotate right  for the specified time at the current rotate_rate
uint32  FF_CMD_SET_DRIVE_RATE=201     # Set drive_rate in M/sec for next drive command
uint32  FF_CMD_SET_ROTATE_RATE=202    # Set the rotation rate for rotate commands in Rad/Sec
uint32  FF_CMD_SET_MAX_LIN_RATE=203   # Set maximum linear rate in M/Sec used to approach the target fiducial
uint32  FF_CMD_SET_MAX_ANG_RATE=204   # Set maximum angular rate in Rad/Sec used to rotate towards the target fiducial

# Actions to take on command done
uint32  FF_ONDONE_DO_NEXT_COMMAND=11  # Default is to go on to next command in the queue
uint32  FF_ONDONE_ASSUME_POSE=12      # Once the fiducial is approached drive on top and rotate to pose of fiducial
uint32  FF_ONDONE_DRIVE_ON_TOP=13     # Drive on top of the fiducial

---
# define the result
uint32  commandResult
---
# define a status
uint32  commandState
uint32  statusBits
float32 percent_complete
